File: src/main/java/com/rayzr522/creativelynamedlib/utils/TextUtils.java
/**
 * 
 */
package com.rayzr522.creativelynamedlib.utils;

import java.util.List;
import java.util.stream.Collectors;

import org.bukkit.ChatColor;

/**
 * Various text-related utilities
 * 
 * @author Rayzr
 */
public class TextUtils {

    /**
     * Translates ampersand (<code>&</code>) color/formatting codes to internal Minecraft color/formatting codes that use the section sign (<code>\u00A7</code>)
     * 
     * @param input The input text to convert
     * @return The converted text
     */
    public static String colorize(String input) {
        return input == null ? null : ChatColor.translateAlternateColorCodes('&', input);
    }

    /**
     * Applies {@link #colorize(String)} to all elements in a {@link List}
     * 
     * @param input The input list of text to convert
     * @return The converted list
     */
    public static List<String> colorize(List<String> input) {
        return input == null ? null : input.stream().map(TextUtils::colorize).collect(Collectors.toList());
    }

    /**
     * Removes all color/formatting codes from the given input
     * 
     * @param input The text to strip formatting from
     * @return The simplified text
     */
    public static String stripColor(String input) {
        return input == null ? null : ChatColor.stripColor(input);
    }

    /**
     * Applies {@link #stripColor(String)} to all elements in a {@link List}
     * 
     * @param input The input list of text to strip formatting from
     * @return The simplified list
     */
    public static List<String> stripColor(List<String> input) {
        return input == null ? null : input.stream().map(TextUtils::stripColor).collect(Collectors.toList());
    }

    /**
     * Translates internal Minecraft color/formatting codes that use the section sign (<code>\u00A7</code>) to ampersand (<code>&</code>) color/formatting codes
     * 
     * @param input The input text to convert
     * @return The converted text
     */
    public static String uncolorize(String input) {
        return input == null ? null : input.replace(ChatColor.COLOR_CHAR, '&');
    }

    /**
     * Applies {@link #uncolorize(String)} to all elements in a {@link List}
     * 
     * @param input The input list of text to convert
     * @return The converted list
     */
    public static List<String> uncolorize(List<String> input) {
        return input == null ? null : input.stream().map(TextUtils::uncolorize).collect(Collectors.toList());
    }

}

File: src/main/kotlin/me/rayzr522/cnl/utils/TextUtils.kt
package me.rayzr522.cnl.utils

import org.bukkit.ChatColor

/**
 * @return This [String] with ampersand color codes translated to use [ChatColor.COLOR_CHAR].
 */
fun String.colorize() = ChatColor.translateAlternateColorCodes('&', this)

/**
 * @return This [String] with [ChatColor.COLOR_CHAR] color codes translated to use ampersands.
 */
fun String.uncolorize() = this.replace(ChatColor.COLOR_CHAR, '&')

/**
 * @return This [String] with all [ChatColor.COLOR_CHAR] color codes removed.
 */
fun String.stripColor() = ChatColor.stripColor(this)


/**
 * @return This list of [String]s with ampersand color codes translated to use [ChatColor.COLOR_CHAR].
 */
fun List<String?>.colorize() = this.map { it?.colorize() }

/**
 * @return This list of [String]s with [ChatColor.COLOR_CHAR] color codes translated to use ampersands.
 */
fun List<String?>.uncolorize() = this.map { it?.uncolorize() }

/**
 * @return This list of [String]s with all [ChatColor.COLOR_CHAR] color codes removed.
 */
fun List<String?>.stripColor() = this.map { it?.stripColor() }